source/development/core-dumps-generation.rst

```diff
@@ -157,4 +157,149 @@ It is up to the developer to decide whether the warning can be ignored or not.
+
+Apport crash files are located in `/var/crash` directory and consist of a package that, not only contains the core dump file, but also process' environment information about the event.
+In order to obtain the specific core dump, the crash report can be unpacked by using `apport-unpack``
+
+  .. code-block:: console
+
+    # cd /var/crash
+    # apport-unpack <dump-filename>.crash <outputdir>
+    # file <outputdir>/Coredump
+    CoreDump: ELF 64-bit LSB core file, x86-64, version 1 (SYSV), SVR4-style, from '/var/ossec/bin/wazuh-logcollector', real uid: 0, effective uid: 0, real gid: 0, effective gid: 0, execfn: '/var/ossec/bin/wazuh-logcollector', platform: 'x86_64'
+
+`<dump-filename>` must be replaced by Apport crash file, that is the full path of the file where slashes (`/`) were replaced by underscores (`_`), plus an incremental counter.
+  For example, first `wazuh-logcollector`` crash will create a report named `_var_ossec_bin_wazuh-logcollector.0.crash`
+
+To know more about Apport see `Apport Wiki <https://wiki.ubuntu.com/Apport>`_.
+
+Direct core dump creation
++++++++++++++++++++++++++
+
+Default on: centOS (5,6,7), OpenSuse.
+
+Several distributions use the simpler and direct mechanism: create the dump on certain directory with a specific name pattern.
+
+.. code-block:: console
+
+    # cat /proc/sys/kernel/core_pattern
+    /tmp/core.%p
+    # file /tmp/core.<pid>
+    CoreDump: ELF 64-bit LSB core file, x86-64, version 1 (SYSV), SVR4-style, from '/var/ossec/bin/wazuh-logcollector', real uid: 0, effective uid: 0, real gid: 0, effective gid: 0, execfn: '/var/ossec/bin/wazuh-logcollector', platform: 'x86_64'
+
+
+Wazuh core dump configuration
++++++++++++++++++++++++++++++
+
+Linux kernel limit the core dump size by default, but need to be extended to obtain a full backtrace.
+Systemd allow us to extend Wazuh service configurations and set this up.
+
+.. code-block:: console
+
+    # mkdir -p /etc/systemd/system/wazuh-<installation>.service.d/
+    # cat > /etc/systemd/system/wazuh-<installation>.service.d/limit_core.conf << EOF
+    [Service]
+    LimitCore=infinity
+    EOF
+    # systemctl daemon-reload
+    # systemctl restart wazuh-<installation>
+
+Wazuh symbols installation
+--------------------------
+
+Debug symbol files will allow the interpretation in a human-readeable way of core dumps. Will be installed in `<INSTALLDIR>/.symbols` directory by default.
+
+.. tabs::
+
+  .. group-tab:: Yum
+
+    .. code-block:: console
+
+      # yum install wazuh-<installation>-debuginfo
+
+  .. group-tab:: APT
+
+    .. code-block:: console
+
+      # apt-get install wazuh-<installation>-dbg
+
+  .. group-tab:: Zypper
+
+    .. code-block:: console
+
+      # zypper install wazuh-<installation>-debuginfo
+
+  .. group-tab:: Installation from sources
+
+    Symbols files will be installed by default on ``<INSTALLDIR>/.symbols`` directory.
+
+  .. group-tab:: WPK installation
+
+    Symbols files are listed :ref:`here<wpk-list>`.
+
```

Comment:
IMHO I would create a new section about how to use GDB with the debugging information in separate files.